daemon/config/config.go

```diff
@@ -408,21 +408,23 @@ func MergeDaemonConfigurations(flagsConfig *Config, flags *pflag.FlagSet, config
 	return fileConfig, nil
 }
 
-// getConflictFreeConfiguration loads the configuration from a JSON file.
-// It compares that configuration with the one provided by the flags,
-// and returns an error if there are conflicts.
+// getConflictFreeConfiguration loads the configuration from a JSON file and
+// applies it to the default configuration. It compares that configuration with
+// the one provided by the flags, and returns an error if there are conflicts.
 func getConflictFreeConfiguration(configFile string, flags *pflag.FlagSet) (*Config, error) {
 	b, err := os.ReadFile(configFile)
 	if err != nil {
 		return nil, err
 	}
 
-	var config Config
+	// Initialize config with the defaults, and apply changes from the JSON file.
+	// TODO(thaJeztah) ideally, this would be a copy of the current config, instead of the default (see Reload() for more details)
+	config := New()
```

Comment:
FWIW (and "note to self") this is the change I want to further look into; this is "fixing" the "try to validate partial configs" part, but I need to verify if it's always doing the right thing.

I guess the better approach would be to not validate until we merged the configs, but for that, we must be sure that we have a copy of the config, and not "live-patch" the daemon's config etc.